scanners/zap/zap.py

```diff
@@ -98,9 +99,25 @@ def postprocess(self):
                 # log path is like '/tmp/rapidast_*/zap.log'
                 tar.add(log, f"evidences/zap_logs/{log.split('/')[-1]}")
 
-        # Calling parent RapidastScanner postprocess
+        self._copy_site_tree()
+
         super().postprocess()
 
+    def _copy_site_tree(self):
+        """
+        Copies the site tree JSON file from the host working directory to the results directory.
+        """
+        site_tree_path = os.path.join(self.host_work_dir, "session_data/zap_site_tree.json")
```

Comment:
like the other comment, zap-site-tree.json.  BTW can we use a variable for the filename?

Reply:
Done! However, storing the filename in a variable adds some complexity to the code. Since ZAP scripts can’t accept custom parameters and this filename is shared between RapiDAST and ZAP.

To pass the filename from RapiDAST to the ZAP script, I used ZAP global variables, to interpolate the script name in Python, I had to use the inline script method. Unfortunately, this only works with the Graal.js engine and doesn’t run with Nashorn. I haven’t looked into this much deeper, as it seems that the Nashorn engine might be already [deprecated](https://github.com/zaproxy/zaproxy/issues/4851) 

Although I could use a single inline script for everything instead of separate scripts for the variable and the main functionality, having the entire script inline can make troubleshooting less user-friendly, especially when using an IDE. Additionally, the inline script was producing strange syntax errors, possibly due to using Graal.js. So, I decided to keep them separate.

Reply:
Not that I know of. Sorry I didn't foresee that it would cause such a trouble. 

In this case, what do you think about reverting the code to your previous one(which is simpler than adding the complexity) but keeping the change from '_' to '-'?

Reply:
My main concern is that we’re hardcoding the filename in two different places. Currently, the two variables remain consistent, but if we revert this code, there's a risk that the filenames could diverge, causing the code to fail if only one of them is changed.

To mitigate this, I can add a comment explaining this dependency and write a unit test to validate that both variables hold the same value. The test would check the JS code using a regular expression to ensure both filenames remain aligned. What do you think?

Reply:
`we’re hardcoding the filename in two different places.` this is why I suggested the variable in the first place too. Anyway, that sounds good.

Reply:
In this case, unless there's a strong objection, I’d prefer to keep the code. Although it introduces some complexity, I believe it’s justified to maintain consistency between both variables. Relying solely on unit tests to enforce this consistency doesn’t seem like the most reliable approach

Reply:
Ok. approved.